src/uv-common.c

```diff
@@ -33,6 +33,33 @@
 # include <net/if.h> /* if_nametoindex */
 #endif
 
+static uv_malloc_func replaced_malloc;
+static uv_free_func replaced_free;
+
+
+void* uv__malloc(size_t size) {
+  if (replaced_malloc)
+    return (*replaced_malloc)(size);
+  return malloc(size);
+}
+
+
+void uv__free(void* ptr) {
+  if (replaced_free)
+    (*replaced_free)(ptr);
+  else
+    free(ptr);
+}
+
+
+void uv_replace_allocator(uv_malloc_func malloc_func, uv_free_func free_func) {
```

Comment:
Release builds can still shoot themselves in the foot. I'd go for making this return an int, and if the allocator was already set we return UV_EINVAL for example.